src/windowscodecs.rs

```diff
@@ -0,0 +1,383 @@
+pub const WICGifGraphicControlExtensionDisposal: ::DWORD = 0x1;
+pub const WICGifGraphicControlExtensionUserInputFlag: ::DWORD = 0x2;
+pub const WICGifGraphicControlExtensionTransparencyFlag: ::DWORD = 0x3;
+pub const WICGifGraphicControlExtensionDelay: ::DWORD = 0x4;
+pub const WICGifGraphicControlExtensionTransparentColorIndex: ::DWORD = 0x5;
+
+pub type WICGifApplicationExtensionProperties = ::DWORD;
+pub const WICGifApplicationExtensionApplication: ::DWORD = 0x1;
+pub const WICGifApplicationExtensionData: ::DWORD = 0x2;
+
+pub type WICGifCommentExtensionProperties = ::DWORD;
+pub const WICGifCommentExtensionText: ::DWORD = 0x1;
+
+pub type WICJpegCommentProperties = ::DWORD;
+pub const WICJpegCommentText: ::DWORD = 0x1;
+
+pub type WICJpegLuminanceProperties = ::DWORD;
+pub const WICJpegLuminanceTable: ::DWORD = 0x1;
+
+pub type WICJpegChrominanceProperties = ::DWORD;
+pub const WICJpegChrominanceTable: ::DWORD = 0x1;
+
+pub type WIC8BIMIptcProperties = ::DWORD;
+pub const WIC8BIMIptcPString: ::DWORD = 0;
+pub const WIC8BIMIptcEmbeddedIPTC: ::DWORD = 0x1;
+
+pub type WIC8BIMResolutionInfoProperties = ::DWORD;
+pub const WIC8BIMResolutionInfoPString: ::DWORD = 0x1;
+pub const WIC8BIMResolutionInfoHResolution: ::DWORD = 0x2;
+pub const WIC8BIMResolutionInfoHResolutionUnit: ::DWORD = 0x3;
+pub const WIC8BIMResolutionInfoWidthUnit: ::DWORD = 0x4;
+pub const WIC8BIMResolutionInfoVResolution: ::DWORD = 0x5;
+pub const WIC8BIMResolutionInfoVResolutionUnit: ::DWORD = 0x6;
+pub const WIC8BIMResolutionInfoHeightUnit: ::DWORD = 0x7;
+
+pub type WIC8BIMIptcDigestProperties = ::DWORD;
+pub const WIC8BIMIptcDigestPString: ::DWORD = 0x1;
+pub const WIC8BIMIptcDigestIptcDigest: ::DWORD = 0x2;
+
+pub type WICPngGamaProperties = ::DWORD;
+pub const WICPngGamaGamma: ::DWORD = 0x1;
+
+pub type WICPngBkgdProperties = ::DWORD;
+pub const WICPngBkgdBackgroundColor: ::DWORD = 0x1;
+
+pub type WICPngItxtProperties = ::DWORD;
+pub const WICPngItxtKeyword: ::DWORD = 0x1;
+pub const WICPngItxtCompressionFlag: ::DWORD = 0x2;
+pub const WICPngItxtLanguageTag: ::DWORD = 0x3;
+pub const WICPngItxtTranslatedKeyword: ::DWORD = 0x4;
+pub const WICPngItxtText: ::DWORD = 0x5;
+
+pub type WICPngChrmProperties = ::DWORD;
+pub const WICPngChrmWhitePointX: ::DWORD = 0x1;
+pub const WICPngChrmWhitePointY: ::DWORD = 0x2;
+pub const WICPngChrmRedX: ::DWORD = 0x3;
+pub const WICPngChrmRedY: ::DWORD = 0x4;
+pub const WICPngChrmGreenX: ::DWORD = 0x5;
+pub const WICPngChrmGreenY: ::DWORD = 0x6;
+pub const WICPngChrmBlueX: ::DWORD = 0x7;
+pub const WICPngChrmBlueY: ::DWORD = 0x8;
+
+pub type WICPngHistProperties = ::DWORD;
+pub const WICPngHistFrequencies: ::DWORD = 0x1;
+
+pub type WICPngIccpProperties = ::DWORD;
+pub const WICPngIccpProfileName: ::DWORD = 0x1;
+pub const WICPngIccpProfileData: ::DWORD = 0x2;
+
+pub type WICPngSrgbProperties = ::DWORD;
+pub const WICPngSrgbRenderingIntent: ::DWORD = 0x1;
+
+pub type WICPngTimeProperties = ::DWORD;

```

Comment:
I was doing all of those like that to be consistent with other parts of winapi-rs, but I'm happy to redo them as enums.